### ISSUE TITLE
Decapitation as a rated victory condition

### ISSUE THREAD

**Issue by phongu** [FAForever/fa#5347]
Decapitation is currently an unrated, modded victory condition implemented by Sheikah. It lets people continue playing even after their commander dies, but the game ends in a loss for the team that loses all commanders.
It's a nice feature for full-share games with a mix of new and veteran players because it avoids the problems associated with the highest rated getting double eco after a beginner dies, and it punishes beginners less for using their commanders, which is otherwise one of the hardest things to learn because of the extreme cost associated with losing it. From the tests I ran with Decapitation, the game balance was less volatile than standard full-share.

Integrating it as a core feature would increase it's visibility (since by default, modded games are not shown in the custom game list) and making it rated has already been ok'd by the balance team in this [forum thread](https://forum.faforever.com/topic/5739/decapitation-should-be-a-rated-victory-condition?_=1692789042117)

**Comment by Garanas** [comment]
We'll pick this up in the 4th developers iteration

**Comment by Garanas** [comment]
I'm not sure if it being integrated would make it rated. I think only assassination is rated. @Askaholic could you help us with that question?

**Comment by Askaholic** [comment]
Yes I believe you’re correct, it would have to be added as a valid victory condition in the server code.

https://github.com/FAForever/server/blob/a6c626afb7b25224988fbe9471c28d8f32a1895f/server/games/game.py#L656

**Comment by Garanas** [comment]
Would that be a difficult change? Bear with me, I have no knowledge of the setup or architecture of the server 😄

**Comment by Askaholic** [comment]
No it wouldn’t be very difficult

**Comment by Garanas** [comment]
Then the remaining question is who decides that the mode is rated or not 😄 , is there previous suggestions or decisions on this topic?

**Comment by Askaholic** [comment]
Great question. I think history won’t really help us here as it probably was determined by a lone dev in the past who just decided how the rating decision should work. I would think some consortium of the balance, game and dev teams might make sense, since it’s kindof more of a metagame decision.

**Comment by Garanas** [comment]
I've started the discussion with the balance team in their channel

**Comment by Garanas** [comment]
The balance team is in favor. That's one down. What other dev teams would be involved in this decision?

**Comment by Askaholic** [comment]
I’d probably see what @Brutus5000 and @Sheikah45 think.

**Comment by Sheikah45** [comment]
I am fine with it. Seems to me to be on the same level as full share vs no share, just has to be clearly communicated which it is in game

**Comment by Garanas** [comment]
I was thinking of an announcement at the start, like the pause announcement?

**Comment by Sheikah45** [comment]
What about just some icon like full vs no share?

Since generally people are annoyed by the text

**Comment by Brutus5000** [comment]
I should not have any stakes in this decision. But I like the idea from a players perspective.

**Comment by Garanas** [comment]
> What about just some icon like full vs no share?
> 
> Since generally people are annoyed by the text

Where is this icon?

**Comment by Garanas** [comment]
I think we can call it and that we can proceed to make it rated. @Askaholic are you available to make the server changes, or should I ask BlackYps?

**Comment by Sheikah45** [comment]
> > What about just some icon like full vs no share?
> > 
> > Since generally people are annoyed by the text
> 
> @Sheikah45 Where is this icon?

It might not be an icon but I am talking about the info in the scoreboard

**Comment by Sheikah45** [comment]
@Garanas One thing I thought of that I never tested with the mod is to make sure units are still transferred when a player gets disconnected.

**Comment by Garanas** [comment]
I'll test it thoroughly when integrating it 😃 . If we put it on the (standard) scoreboard then people may not see it because of UI mods replacing the scoreboard

**Comment by Sheikah45** [comment]
I would probably be inclined to say that is on the scoreboard author to update it to reflect the information. As historically by now we have established like info like that is on the scoreboard. But that is just my opinion.

Rather than having a text announcement every game.

**Comment by Garanas** [comment]
> I would probably be inclined to say that is on the scoreboard author to update it to reflect the information. As historically by now we have established like info like that is on the scoreboard. But that is just my opinion.

Fair enough, I agree

**Comment by BlackYps** [comment]
> Yes I believe you’re correct, it would have to be added as a valid victory condition in the server code.
> 
> https://github.com/FAForever/server/blob/a6c626afb7b25224988fbe9471c28d8f32a1895f/server/games/game.py#L656

So this is supposed to get integrated into the base game? I believe the relevant lines would then look like this
```python
if self.gameOptions["Victory"] not in (Victory.DEMORALIZATION, Victory.DECAPITATION):
    await self.mark_invalid(ValidityState.WRONG_VICTORY_CONDITION)

**Comment by Garanas** [comment]
And where does it get the `DECAPITATION` value from?

**Comment by BlackYps** [comment]
On the server the victory is just an enum that matches the victory conditions to integers. I am not sure where those originate from, but they also seem to be saved in the database.

**Comment by Garanas** [comment]
I'll soon start on implementing this so that we can ship it with the 4th development iteration. Has there been work on the server on the matter?

**Comment by BlackYps** [comment]
No, not yet

**Comment by Askaholic** [comment]
The server and database work for this is done.
